### PR TITLE
Upgrade golang version in Actions and set golang version aotomatically

### PR DESCRIPTION
**Pull request by junw-33455432**

### CODE REVIEW

**Review by Xiaochao8 — CHANGES_REQUESTED** [review]
```
RestAPI
Total tests run: 32, Failures: 20, Skips: 0
```

Smoke test failed but reported as pass. Please check.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ jobs:
       - name: Coverage
         run: |
           make coverage
-          sed -i "/^.*internal\/bindata\/cldr\/bindata.go.*$/d" ./cover.out
+          sed -i "/^.*internal\/bindata\/cldr.*$/d" ./cover.out
```
Comment (Xiaochao8) [inline .github/workflows/main.yml:78]: Please ignore the folder `internal/bindata`

**Review by Xiaochao8 — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (junw-33455432) [reply .github/workflows/main.yml]: Updated.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -75,7 +75,7 @@ jobs:
       - name: Coverage
         run: |
           make coverage
-          sed -i "/^.*internal\/bindata\/cldr.*$/d" ./cover.out
+          sed -i "/^.*internal\/bindata.*$/d" ./cover.out
```
Comment (Xiaochao8) [inline .github/workflows/main.yml:78]: bindata is a folder. Ignore anything in it.
```suggestion
          sed -i "/^.*internal\/bindata\/.*$/d" ./cover.out
```

**Review by Xiaochao8 — CHANGES_REQUESTED** [review]
There are errors in smoke test, but CI reports the step as pass. Please check.

`Total tests run: 32, Failures: 20, Skips: 0`

**Comment by junw-33455432** [comment]
Merged the PR according to @Xiaochao8 's comments, close this PR, @Xiaochao8 will fix the Unit Test failure in another PR.

**Pull request merged**